### PR TITLE
DRAFT: feat: add robots.txt w/ auto update

### PR DESCRIPTION
**Pull request by mrshmllow**
I will not open for review until https://forum.aux.computer/t/anti-ai-llm-embargo/425 passes

- Automatically checks DarkVisitors for updates to `public/robots.txt` every week
- Creates a PR if there are any changes.
- Requires a `DarkVisitorsBearer` repo secret.

### CODE REVIEW

File: .github/workflows/robots.txt.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+name: "Update robots.txt"
+on:
+  schedule:
+  - cron: "0 0 * * 6"
```
Comment (isabelroses) [inline .github/workflows/robots.txt.yml]:
```suggestion
  - cron: "0 0 * * 6" # At 00:00 on Saturday
```

File: .github/workflows/robots.txt.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+            git push --force-with-lease -u origin "$PR_BRANCH"
+          fi
+
+      - name: Create PR
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          gh pr create \
+              --base "$CURRENT_BRANCH" --head "$PR_BRANCH" --title "Update robots.txt $(date '+%Y %m %d')" --fill
```
Comment (isabelroses) [inline .github/workflows/robots.txt.yml]: I think we can simplify this by just having the ci push to main.
```suggestion
    - name: Commit changes
       uses: EndBug/add-and-commit@v9
       with:
          message: "chore: generate robots.txt"
          default_author: github_actions
          push: true
```

File: .github/workflows/robots.txt.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: write
+      pull-requests: write
+
+    env:
+      CURRENT_BRANCH: ${{ github.ref_name }}
+      PR_BRANCH: update-robots-${{ github.ref_name }}
```
Comment (isabelroses) [inline .github/workflows/robots.txt.yml]:
```suggestion
```
re: ci push

File: .github/workflows/robots.txt.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+jobs:
+  update:
+    name: "Update"
+
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: write
+      pull-requests: write
```
Comment (isabelroses) [inline .github/workflows/robots.txt.yml]:
```suggestion
```
re: ci push

File: .github/workflows/robots.txt.yml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        uses: actions/checkout@v4
+
+      - name: Setup Git user info
+        run: |
+          git config user.name 'github-actions[bot]'
+          git config user.email 'github-actions[bot]@users.noreply.github.com'
+
+      - name: Create new branch
+        run: |
+          git switch -c "$PR_BRANCH"
```
Comment (isabelroses) [inline .github/workflows/robots.txt.yml]:
```suggestion
```
re: ci push

File: .github/workflows/robots.txt.yml [hunk 6]
```diff
@@ -0,0 +1,60 @@
+name: "Update robots.txt"
+on:
+  schedule:
+  - cron: "0 0 * * 6"
+  workflow_dispatch:
+
+jobs:
+  update:
+    name: "Update"
```
Comment (isabelroses) [inline .github/workflows/robots.txt.yml]:
```suggestion
    name: "Update robots.txt"
```

**Comment by isabelroses** [comment]
> * Requires a `DarkVisitorsBearer` repo secret.

I have added the secret in preparation.

File: .github/workflows/robots.txt.yml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Pull latest robots
+        run: |
+          curl -X POST https://api.darkvisitors.com/robots-txts \
+              -H "Authorization: Bearer ${{ secrets.DarkVisitorsBearer }}" \
+              -H "Content-Type: application/json" \
+              --data-raw '{"agent_types": ["AI Assistant", "AI Data Scraper", "AI Search Crawler", "Undocumented AI Agent"]}' \
```
Comment (mrshmllow) [inline .github/workflows/robots.txt.yml:25]: I wish the service defaulted to ALL on this key..

**Comment by isabelroses** [comment]
> I will not open for review until [forum.aux.computer/t/anti-ai-llm-embargo/425](https://forum.aux.computer/t/anti-ai-llm-embargo/425) passes

The vote passed with an overwhelming majority. 98% for, so we are good to merge.

**Review by isabelroses — APPROVED** [review]

**Pull request merged**